# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../dansguardian/gcc44.patch
# Copyright (C) 2010 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

http://sources.gentoo.org/cgi-bin/viewvc.cgi/gentoo-x86/net-proxy/dansguardian/files/dansguardian-2.10.1.1-gcc44.patch?view=markup

diff -Nru dansguardian-2.10.1.1.orig/src/ConnectionHandler.cpp dansguardian-2.10.1.1/src/ConnectionHandler.cpp
--- dansguardian-2.10.1.1.orig/src/ConnectionHandler.cpp	2009-02-25 12:36:22.000000000 +0100
+++ dansguardian-2.10.1.1/src/ConnectionHandler.cpp	2009-07-05 01:15:52.000000000 +0200
@@ -44,6 +44,7 @@
 #include <istream>
 
 #ifdef ENABLE_ORIG_IP
+#include <limits.h>
 #include <linux/types.h>
 #include <linux/netfilter_ipv4.h>
 #endif
diff -Nru dansguardian-2.10.1.1.orig/src/downloadmanagers/fancy.cpp dansguardian-2.10.1.1/src/downloadmanagers/fancy.cpp
--- dansguardian-2.10.1.1.orig/src/downloadmanagers/fancy.cpp	2008-11-18 12:27:04.000000000 +0100
+++ dansguardian-2.10.1.1/src/downloadmanagers/fancy.cpp	2009-07-05 01:15:12.000000000 +0200
@@ -26,6 +26,7 @@
 #include "../HTMLTemplate.hpp"
 #include "../ConnectionHandler.hpp"
 
+#include <stdio.h>
 #include <syslog.h>
 #include <sys/time.h>
 #include <sys/types.h>
diff -Nru dansguardian-2.10.1.1.orig/src/String.cpp dansguardian-2.10.1.1/src/String.cpp
--- dansguardian-2.10.1.1.orig/src/String.cpp	2009-01-16 11:44:28.000000000 +0100
+++ dansguardian-2.10.1.1/src/String.cpp	2009-07-05 01:15:12.000000000 +0200
@@ -164,9 +164,9 @@
 		return 0;
 	off_t t = 0;
 	if (sizeof(off_t) == 4)
-		sscanf(this->c_str(), "%d", &t);
+		sscanf(this->c_str(), "%d", (long *)&t);
 	else if (sizeof(off_t) == 8)
-		sscanf(this->c_str(), "%lld", &t);
+		sscanf(this->c_str(), "%lld", (long long*)&t);
 	return t;
 }

